x/auth/keeper/genesis.go

```diff
@@ -28,6 +28,14 @@ func (ak AccountKeeper) InitGenesis(ctx sdk.Context, data types.GenesisState) {
 			n := ak.NextAccountNumber(ctx)
 			lastAccNum = &n
 		}
+
+		// TODO HV2: this is imported from heimdall
+		baseAcc := types.NewBaseAccount(acc.GetAddress(), acc.GetPubKey(), accNum, acc.GetSequence())
+
+		// execute account processors
+		for _, p := range processors {
+			acc = p(&acc, baseAcc) //nolint
```

Comment:
There is a subtle difference here w.r.t. Heimall.
* Here we are passing the `acc` as it was returned in the previous iteration of the inner loop
* whereas, in Heimdall, we are passing `gacc`, which is external to this inner loop, so the same in all iterations

Reply:
Oh, thanks for noticing it. Changed. Will push soon.

Reply:
Resolved